compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -1235,18 +1235,27 @@ object desugar {
       rhsOK(rhs)
   }
 
+  def checkOpaqueAlias(tree: MemberDef)(using Context): MemberDef =
+    if !tree.mods.is(Opaque) then tree
+    else tree match
+      case TypeDef(_, bounds: TypeBoundsTree) if bounds.alias.isEmpty =>
+        report.error(i"opaque type must have a right-hand side", tree.srcPos)
```

Comment:
```suggestion
        report.error(i"opaque type must have a right-hand side", tree.srcPos.endPos)
```

...if we want the caret to be placed where the RHS should go?

Reply:
Both approaches seem reasonable to me. I do not have a preference on either option.